core/trino-main/src/main/java/io/trino/cost/ComparisonStatsCalculator.java

```diff
@@ -239,4 +252,134 @@ private static PlanNodeStatsEstimate estimateExpressionNotEqualToExpression(
+        // to make a conservative estimate for nullsFilterFactor
+        double nullsFilterFactor;
+        if (isNaN(leftExpressionStatistics.getNullsFraction()) && isNaN(rightExpressionStatistics.getNullsFraction())) {
+            nullsFilterFactor = 1.0;
+        }
+        else {
+            nullsFilterFactor = 1 - maxExcludeNaN(leftExpressionStatistics.getNullsFraction(), rightExpressionStatistics.getNullsFraction());
+        }
+
+        switch (operator) {
+            case LESS_THAN:
+            case LESS_THAN_OR_EQUAL:
+                // left is always greater than right, no overlap
+                if (leftRange.getLow() > rightRange.getHigh()) {
+                    return inputStatistics.mapOutputRowCount(rowCount -> 0.0);
+                }
+                // left is always lesser than right
+                if (leftRange.getHigh() < rightRange.getLow()) {
+                    PlanNodeStatsEstimate.Builder estimate = PlanNodeStatsEstimate.buildFrom(inputStatistics);
+                    leftExpressionSymbol.ifPresent(symbol -> estimate.addSymbolStatistics(
+                            symbol,
+                            leftExpressionStatistics.mapNullsFraction(nullsFraction -> 0.0)));
+                    rightExpressionSymbol.ifPresent(symbol -> estimate.addSymbolStatistics(
+                            symbol,
+                            rightExpressionStatistics.mapNullsFraction(nullsFraction -> 0.0)));
+                    return estimate.setOutputRowCount(inputStatistics.getOutputRowCount() * nullsFilterFactor)
+                            .build();
+                }
+                if (isNonEstimatablePredicateApproximationEnabled) {
+                    PlanNodeStatsEstimate.Builder estimate = PlanNodeStatsEstimate.buildFrom(inputStatistics);
+                    double overlappingRangeFraction = leftRange.overlapPercentWith(leftRange.intersect(rightRange));
+                    double alwaysLessRangeFraction = 0.0;
+                    if (leftRange.getLow() < rightRange.getLow()) {
+                        alwaysLessRangeFraction = leftRange.overlapPercentWith(new StatisticRange(leftRange.getLow(), rightRange.getLow(), NaN));
+                    }
+                    double filterFactor = alwaysLessRangeFraction + (OVERLAPPING_RANGE_INEQUALITY_FILTER_COEFFICIENT * overlappingRangeFraction);
+                    leftExpressionSymbol.ifPresent(symbol -> estimate.addSymbolStatistics(
+                            symbol,
+                            SymbolStatsEstimate.builder()
+                                    .setLowValue(leftRange.getLow())
+                                    .setHighValue(minExcludeNaN(leftRange.getHigh(), rightRange.getHigh()))
+                                    .setAverageRowSize(leftExpressionStatistics.getAverageRowSize())
+                                    .setDistinctValuesCount(leftExpressionStatistics.getDistinctValuesCount() * filterFactor)
```

Comment:
I don't think NDV count is reduced apart from `alwaysGreaterRange`
1. in `alwaysLessRange` no NDVs are filtered
2. in overlapping range, each NDV should be part of at least one pair